diff -ru glib-1.2.8.orig/configure.in glib-1.2.8/configure.in
--- glib-1.2.8.orig/configure.in	2005-06-26 20:38:20.468504000 +0200
+++ glib-1.2.8/configure.in	2005-06-26 20:40:14.057549097 +0200
@@ -336,7 +336,7 @@
 if test $gtk_ok = yes; then
     AC_MSG_RESULT([yes, found in sys/types.h])
 else
-    AC_HEADER_EGREP(fd_mask, sys/select.h, gtk_ok=yes)
+    AC_HEADER_EGREP(fd_set, sys/select.h, gtk_ok=yes)
     if test $gtk_ok = yes; then
         AC_DEFINE(HAVE_SYS_SELECT_H)
         AC_MSG_RESULT([yes, found in sys/select.h])
Only in glib-1.2.8: configure.in~
