--- glib-1.2.8/tests/Makefile.am.old	2005-03-28 13:18:58.219623027 +0200
+++ glib-1.2.8/tests/Makefile.am	2005-03-28 13:19:07.243153444 +0200
@@ -34,3 +34,5 @@
 
 makefile.msc: $(top_builddir)/config.status $(top_srcdir)/tests/makefile.msc.in
 	cd $(top_builddir) && CONFIG_FILES=tests/$@ CONFIG_HEADERS= $(SHELL) ./config.status
+
+CLEANFILES = makefile.msc
--- glib-1.2.8/Makefile.am.old	2005-03-28 13:20:29.053829656 +0200
+++ glib-1.2.8/Makefile.am	2005-03-28 13:20:47.429836916 +0200
@@ -120,3 +120,8 @@
 	     && cp INSTALL README $(distdir) ; \
 	fi \
 	&& cp glib.spec $(distdir)
+
+DISTCLEANFILES = glibconfig-sysdefs.h glibconfig.h glibconfig.h.win32 \
+               config.h.win32
+
+CLEANFILES = stamp-gc-h  makefile.msc
