diff --git a/glib/glib/Makefile.am b/glib/glib/Makefile.am
index 07b0141..8edc404 100644
--- a/glib/glib/Makefile.am
+++ b/glib/glib/Makefile.am
@@ -12,7 +12,7 @@ include $(top_srcdir)/glib.mk
 DISTCLEANFILES += glibconfig-stamp glibconfig.h
 BUILT_SOURCES += glibconfig-stamp
 configexecincludedir = $(libdir)/glib-2.0/include
-nodist_configexecinclude_HEADERS = glibconfig.h
+nodist_noinst_HEADERS = glibconfig.h
 glibconfig-stamp: ../config.status
 	$(AM_V_GEN) cd $(top_builddir) && \
 	  $(SHELL) ./config.status glib/glibconfig.h
@@ -60,18 +60,15 @@ EXTRA_DIST += 			\
 	gregex.c		\
 	gregex.h		\
 	win_iconv.c		\
-	libglib-gdb.py.in	\
 	docs.c			\
 	gconstructor.h		\
 	$(MIRRORING_TAB_SOURCE)
 
-CLEANFILES += libglib-gdb.py
-
 # These may be in the builddir too
 BUILT_EXTRA_DIST += 		\
 	glib.rc
 
-lib_LTLIBRARIES = libglib-2.0.la
+noinst_LTLIBRARIES += libglib-2.0.la
 
 if OS_WIN32_AND_DLL_COMPILATION
 if MS_LIB_AVAILABLE
@@ -220,13 +217,13 @@ EXTRA_libglib_2_0_la_SOURCES = \
 	gwin32.c
 
 glibincludedir=$(includedir)/glib-2.0
-glibinclude_HEADERS =   \
+noinst_HEADERS =   \
 	glib-unix.h     \
 	glib-object.h	\
 	glib.h
 
 deprecatedincludedir=$(includedir)/glib-2.0/glib/deprecated
-deprecatedinclude_HEADERS = \
+noinst_HEADERS += \
 	deprecated/gallocator.h		\
 	deprecated/gcache.h		\
 	deprecated/gcompletion.h	\
@@ -235,7 +232,7 @@ deprecatedinclude_HEADERS = \
 	deprecated/gthread.h
 
 glibsubincludedir=$(includedir)/glib-2.0/glib
-glibsubinclude_HEADERS = \
+noinst_HEADERS += \
 	galloca.h	\
 	garray.h	\
 	gasyncqueue.h	\
@@ -317,15 +314,6 @@ CLEANFILES += glib-public-headers.txt
 
 all-local: glib-public-headers.txt
 
-install-data-local: install-ms-lib
-	@if test -f $(glibincludedir)/glist.h ; then					\
-	  echo "*** Old headers found in $(glibincludedir). You should remove the" ;	\
-	  echo "*** contents of this directory and type 'make install' again." ;	\
-	  false ;									\
-	fi
-
-uninstall-local: uninstall-ms-lib uninstall-gdb
-
 if PLATFORM_WIN32
 no_undefined = -no-undefined
 endif
@@ -349,11 +337,8 @@ libglib_2_0_la_DEPENDENCIES = libcharset/libcharset.la $(printf_la) @GIO@ @GSPAW
 
 libglib_2_0_la_LDFLAGS = $(GLIB_LINK_FLAGS) \
 	 $(glib_win32_res_ldflag) \
-	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-export-dynamic $(no_undefined)
 
-INSTALL_PROGS=
-
 gspawn-win32-helper-console.c:
 	echo '#define HELPER_CONSOLE' >$@
 	echo '#include "gspawn-win32-helper.c"' >>$@
@@ -366,50 +351,17 @@ gspawn-win64-helper-console.c:
 	echo '#include "gspawn-win32-helper.c"' >>$@
 
 
-if OS_WIN32
-if OS_WIN32_X64
-INSTALL_PROGS += gspawn-win64-helper gspawn-win64-helper-console
-gspawn_win64_helper_LDADD = libglib-2.0.la
-gspawn_win64_helper_LDFLAGS = -mwindows
-gspawn_win64_helper_console_LDADD = libglib-2.0.la
-else
-INSTALL_PROGS += gspawn-win32-helper gspawn-win32-helper-console
-gspawn_win32_helper_LDADD = libglib-2.0.la
-gspawn_win32_helper_LDFLAGS = -mwindows
-gspawn_win32_helper_console_LDADD = libglib-2.0.la
-endif
-endif
-
 glib-win32-res.o: glib.rc
 	$(WINDRES) glib.rc $@
 
-bin_PROGRAMS	 = ${INSTALL_PROGS}
-
 if OS_UNIX
 
-INSTALL_PROGS	+= gtester
-gtester_SOURCES	 = gtester.c
-gtester_LDADD	 = libglib-2.0.la 
-
 auto_config_binscripts = gtester-report
-bin_SCRIPTS = ${auto_config_binscripts}
 EXTRA_DIST += ${auto_config_binscripts}
 
 CONFIGVARS = \
 	"bindir"    	: "${bindir}",		\
 	"glib-version"  : "${GLIB_VERSION}"
-
-install-exec-hook:
-	for sf in ${auto_config_binscripts} ; do \
-	  mv -f "$(DESTDIR)$(bindir)/$$sf" "$(DESTDIR)$(bindir)/$$sf".tmp \
-	  && sed < "$(DESTDIR)$(bindir)/$$sf".tmp > "$(DESTDIR)$(bindir)/$$sf" \
-	    -e '1,24s|^ *#@PKGINSTALL_CONFIGVARS_IN24LINES@|  ${CONFIGVARS}|' \
-	    -e '1,1s|#! /usr/bin/env python.*|#!${PYTHON}|' \
-	  || exit $$? ; \
-	  chmod a+x $(DESTDIR)$(bindir)/$$sf ; \
-	  rm -f "$(DESTDIR)$(bindir)/$$sf".tmp ; \
-	done
-
 endif
 
 glib-2.0.lib: libglib-2.0.la glib.def
@@ -456,25 +408,3 @@ dist-hook: $(BUILT_EXTRA_DIST)
 	done >libglib.vs10.sourcefiles.filters
 	$(CPP) -P - <$(top_srcdir)/build/win32/vs10/glib.vcxproj.filtersin >$@	
 	rm libglib.vs10.sourcefiles.filters
-
-# install gdb scripts
-gdbdir = $(datadir)/glib-2.0/gdb
-dist_gdb_SCRIPTS = glib.py
-
-libglib-gdb.py: libglib-gdb.py.in
-	$(AM_V_GEN) $(SED) -e "s|\@datadir\@|$(datadir)|" $(srcdir)/libglib-gdb.py.in > $(builddir)/libglib-gdb.py
-
-
-install-data-hook: libglib-gdb.py
-	mkdir -p $(DESTDIR)$(datadir)/gdb/auto-load$(ABS_GLIB_RUNTIME_LIBDIR)
-	$(INSTALL) $(builddir)/libglib-gdb.py $(DESTDIR)$(datadir)/gdb/auto-load$(ABS_GLIB_RUNTIME_LIBDIR)/libglib-2.0.so.0.$(LT_CURRENT).$(LT_REVISION)-gdb.py
-if HAVE_GLIB_RUNTIME_LIBDIR
-	mkdir -p $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	mv $(DESTDIR)$(libdir)/libglib-2.0.so.0 $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	mv $(DESTDIR)$(libdir)/libglib-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/$(GLIB_RUNTIME_LIBDIR)
-	rm -f $(DESTDIR)$(libdir)/libglib-2.0.so
-	ln -s $(GLIB_RUNTIME_LIBDIR)/libglib-2.0.so.0.$(LT_CURRENT).$(LT_REVISION) $(DESTDIR)$(libdir)/libglib-2.0.so
-endif
-
-uninstall-gdb:
-	-rm -r $(DESTDIR)$(datadir)/gdb
diff --git a/glib/glib/libcharset/Makefile.am b/glib/glib/libcharset/Makefile.am
index 642f75c..abf32cb 100644
--- a/glib/glib/libcharset/Makefile.am
+++ b/glib/glib/libcharset/Makefile.am
@@ -26,31 +26,6 @@ EXTRA_DIST += 			\
 
 charset_alias = $(DESTDIR)$(libdir)/charset.alias
 charset_tmp = $(DESTDIR)$(libdir)/charset.tmp
-install-exec-local: all-local
-	$(mkinstalldirs) $(DESTDIR)$(libdir)
-	if test -f $(charset_alias); then \
-	  sed -f ref-add.sed $(charset_alias) > $(charset_tmp) ; \
-	  $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
-	  rm -f $(charset_tmp) ; \
-	else \
-	  if test @GLIBC21@ = no; then \
-	    sed -f ref-add.sed charset.alias > $(charset_tmp) ; \
-	    $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
-	    rm -f $(charset_tmp) ; \
-	  fi ; \
-	fi
-
-uninstall-local: all-local
-	if test -f $(charset_alias); then \
-	  sed -f ref-del.sed $(charset_alias) > $(charset_tmp); \
-	  if grep '^# Packages using this file: $$' $(charset_tmp) \
-	      > /dev/null; then \
-	    rm -f $(charset_alias); \
-	  else \
-	    $(INSTALL_DATA) $(charset_tmp) $(charset_alias); \
-	  fi; \
-	  rm -f $(charset_tmp); \
-	fi
 
 charset.alias: config.charset
 	$(AM_V_GEN) $(SHELL) $(srcdir)/config.charset '@host@' > t-$@
diff --git a/glib/m4macros/Makefile.am b/glib/m4macros/Makefile.am
index 971871c..c91daa8 100644
--- a/glib/m4macros/Makefile.am
+++ b/glib/m4macros/Makefile.am
@@ -5,4 +5,4 @@ installed_m4= glib-2.0.m4 glib-gettext.m4 gsettings.m4
 EXTRA_DIST+=$(installed_m4)
 
 m4datadir = $(datadir)/aclocal
-m4data_DATA = $(installed_m4)
+noinst_DATA += $(installed_m4)
