diff --git a/glib/configure.ac b/glib/configure.ac
index 67fa762..5495c75 100644
--- a/glib/configure.ac
+++ b/glib/configure.ac
@@ -485,6 +485,11 @@ case $with_libiconv in
     ;;
 esac
 
+dnl Ensure we do a static-only build. This really only matters on Windows,
+dnl but it's safer to do it for all platforms.
+enable_static=yes
+enable_shared=no
+
 dnl Initialize libtool
 LT_PREREQ([2.2])
 LT_INIT([disable-static win32-dll])
